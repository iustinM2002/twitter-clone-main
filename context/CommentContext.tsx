import React,{useState,createContext} from 'react';
// interface for initialValue
interface CommentInit{
  activeComment:[
    activeCom:Boolean,
    setActiveCom:  React.Dispatch<React.SetStateAction<boolean>>
  ],
  commentLib:[
    comment:string,
    setComment: React.Dispatch<React.SetStateAction<string>>
  ],
  commentsLib:[
    comments:string[],
    setComments: React.Dispatch<React.SetStateAction<string[]>>
  ],
  replyUser:[
    replyedUsername:string[],
    setReplyedUsername: React.Dispatch<React.SetStateAction<string[]>>
  ]
}
// create context for comments
let initValue!:CommentInit;
export const CommentContext = createContext(initValue);

export const CommentProvider = (props:any) => {
  // state declarations
    const [activeCom,setActiveCom] = useState(false);
    const [comment,setComment] = useState<string>('');
    const [comments,setComments] = useState<string[]>([]);
    const [replyedUsername,setReplyedUsername] = useState<string[]>([]);

  return (
    <CommentContext.Provider value={{activeComment:[activeCom,setActiveCom],commentLib:[comment,setComment],commentsLib:[comments,setComments],replyUser:[replyedUsername,setReplyedUsername]}}>
        {props.children}
    </CommentContext.Provider>
  )
}

