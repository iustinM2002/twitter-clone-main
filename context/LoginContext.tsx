import React,{ createContext,useReducer, useState} from 'react';
// importing reducers
import { initalActive,RegisterActiveReducer } from '../reducers/ActiveReducer';

interface InitialInt  {
    registerFormSwitch:[
        ActiveRegister:Boolean,
        dispatch: any
    ],
    switchLogged:[
      isLogged:string,
      dispatchLog: React.Dispatch<React.SetStateAction<string>> | any
    ]
}
let initialValue!:InitialInt;

export const LoginContext = createContext(initialValue);
export const LoginProvider = (props:any) => {
    const [ActiveRegister,dispatch] = useReducer(RegisterActiveReducer,initalActive);
    const [isLogged,dispatchLog] = useState('false')
  return (
    <LoginContext.Provider value={{registerFormSwitch:[ActiveRegister,dispatch],switchLogged:[isLogged,dispatchLog]}}>
        {props.children}
    </LoginContext.Provider>
  )
}