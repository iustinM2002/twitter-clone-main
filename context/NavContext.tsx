import React, { createContext,useState } from 'react';
type navInt =  [  
   navIndex:number ,
    setNavIndex:React.Dispatch<React.SetStateAction<number>>[] | any,
]
let initValue!:navInt ;
export const NavContext = createContext(initValue)

export const NavProvider = (props:any) => {
    const [navIndex,setNavIndex] =useState<number>(0)
  return (
    <NavContext.Provider value={[navIndex,setNavIndex]}>{props.children}</NavContext.Provider>
  )
}

