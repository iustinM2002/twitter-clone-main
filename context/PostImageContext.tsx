import React,{useReducer,createContext,useState} from 'react';
import {initialImages,ImageReducer} from '../reducers/ImageReducer'
interface initalValue{
    ImageSrc:string[] | any,
    dispatchImg:any
}
let initValue!:initalValue | any;
export const PostImageContext = createContext(initValue)

export const PostImageProvider = (props:any) => {
    const [ImagesSrc,dispatchImg] = useReducer(ImageReducer,initialImages)
    const [ImageSrc,setImageSrc] = useState('')
  return (
    <PostImageContext.Provider value={[ImageSrc,setImageSrc,ImagesSrc,dispatchImg]}>{props.children}</PostImageContext.Provider>
  )
}