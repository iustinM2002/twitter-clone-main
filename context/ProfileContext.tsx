import React,{createContext,useState,useReducer} from 'react';


interface initialValue{
    currentPicture:[
        currentProfilePic:string,
        setCurrentProfilePic:any
    ],
    currentUsername:[
        currentUserName:string,
        setCurrentUsername:any
    ],
    currentBgPic:[
      currentBackgroundPic:string,
      setCurrentBackgroundPic:any
    ],
    currentSelectedUser:[
      currentUser:{},
      setCurrentUser:any
    ],
}

let initValue!:initialValue;
export const ProfileContext = createContext(initValue)

export const ProfileProvider = (props:any) => {
    const [currentProfilePic,setCurrentProfilePic] = useState<string>('https://twirpz.files.wordpress.com/2015/06/twitter-avi-gender-balanced-figure.png');
    const [currentBackgroundPic,setCurrentBackgroundPic] = useState<string>('https://www.comicmoviedb.com/wp-content/uploads/2016/01/twitter-bg.jpg')
    const [currentUserName,setCurrentUsername] = useState<string>('username');
    const [currentUser,setCurrentUser] = useState<{}>({})
  return (
    <ProfileContext.Provider value={{currentPicture:[currentProfilePic,setCurrentProfilePic],currentUsername:[currentUserName,setCurrentUsername],currentBgPic:[currentBackgroundPic,setCurrentBackgroundPic],currentSelectedUser:[currentUser,setCurrentUser]}}>{props.children}</ProfileContext.Provider>
  )
}

