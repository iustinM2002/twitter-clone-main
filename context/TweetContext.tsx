import React,{createContext,useReducer,useState} from 'react';
import {TweetReducer,initialTweet} from '../reducers/TweetReducer';

type initialValue = [
    addTweets:string[] | any,
    dispatchTweets:any,
    tweet:string,
    setTweet:any
]

let initValue!:initialValue;
export const TweetContext = createContext(initValue);

export const TweetProvider = (props:any) => {
    const [addTweets,dispatchTweets] = useReducer(TweetReducer,initialTweet)
    const [tweet,setTweet] = useState<string>('')
  return (
    <TweetContext.Provider value={[addTweets,dispatchTweets,tweet,setTweet]} >
        {props.children}
    </TweetContext.Provider>
  )
}

export default TweetProvider