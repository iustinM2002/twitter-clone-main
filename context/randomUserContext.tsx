import React,{createContext,useEffect,useState} from 'react';
import { QueryClient,dehydrate,useQuery } from 'react-query';
import Loading from '../components/utils/Loading'
type initalValue = any;
let initValue!:initalValue;
export const RandomContext = createContext(initValue)
let index = 50;

 const randomUsers= async () => await(await fetch(`https://randomuser.me/api/?results=${index}`)).json()
 const randomIpsum= async () => await(await fetch('https://random-data-api.com/api/lorem_ipsum/random_lorem_ipsum?size=50')).json()
 const randomCompany= async () => await(await fetch('https://random-data-api.com/api/company/random_company?size=3')).json()
 const randomPost = async () => await( await fetch('https://dummyapi.io/data/v1/post?limit=50',{headers:{'app-id':'625ffa6de6a875ce42705773',}})).json();
 const randomPicsum = async () => await(await fetch('https://picsum.photos/v2/list?page=2&limit=50')).json()
 const randomTag = async () => await( await fetch('https://dummyapi.io/data/v1/tag?limit=1',{headers:{'app-id':'625ffa6de6a875ce42705773',}})).json();
 
export const RandomUserProvider = (props:any) => {
  const randomUsersQuery = useQuery('random_users', randomUsers);
  const randomIpsumQuery = useQuery('random_ipsum', randomIpsum);
  const randomCompanyQuery = useQuery('random_compnay', randomCompany);
  const randomPostQuery = useQuery('random_post',randomPost);
  const randomPicsumQuery = useQuery('random_picusm',randomPicsum);
  const randomTagQuery = useQuery('random_tag',randomTag);
  

  if(randomIpsumQuery.isLoading || randomUsersQuery.isLoading || randomCompanyQuery.isLoading || randomPostQuery.isLoading || randomPicsumQuery.isLoading || randomTagQuery.isLoading){
    return <Loading/>
  }

  return (
    <RandomContext.Provider value={[randomUsersQuery,randomIpsumQuery,randomCompanyQuery,randomPostQuery,randomPicsumQuery,randomTagQuery]}>{props.children}</RandomContext.Provider>
  )
}

