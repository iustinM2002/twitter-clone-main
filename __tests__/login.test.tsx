import { render } from "@testing-library/react";
import Home from '../pages/index';


import { LoginProvider } from '../context/LoginContext';
test('homepage match snapshot',() =>{
    const { container } = render(<LoginProvider><Home initialContacts={[]}/> </LoginProvider>)
    expect(container).toMatchSnapshot()
    
})