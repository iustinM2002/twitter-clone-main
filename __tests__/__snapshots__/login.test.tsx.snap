// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`homepage match snapshot 1`] = `
<div>
  <div>
    <div
      class="min-h-[100vh] w-full flex lg:flex-col"
    >
      
      <div
        class="image-starter min-h-[100vh] w-[50%] bg-cover bg-center bg-[url('../images/starter_semi_bg.png')] flex justify-center items-center lg:order-2 lg:w-full lg:min-h-[50vh]"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-twitter text-white text-[15rem] sm:text-[10rem]"
          data-icon="twitter"
          data-prefix="fab"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M459.4 151.7c.325 4.548 .325 9.097 .325 13.65 0 138.7-105.6 298.6-298.6 298.6-59.45 0-114.7-17.22-161.1-47.11 8.447 .974 16.57 1.299 25.34 1.299 49.06 0 94.21-16.57 130.3-44.83-46.13-.975-84.79-31.19-98.11-72.77 6.498 .974 12.99 1.624 19.82 1.624 9.421 0 18.84-1.3 27.61-3.573-48.08-9.747-84.14-51.98-84.14-102.1v-1.299c13.97 7.797 30.21 12.67 47.43 13.32-28.26-18.84-46.78-51.01-46.78-87.39 0-19.49 5.197-37.36 14.29-52.95 51.65 63.67 129.3 105.3 216.4 109.8-1.624-7.797-2.599-15.92-2.599-24.04 0-57.83 46.78-104.9 104.9-104.9 30.21 0 57.5 12.67 76.67 33.14 23.72-4.548 46.46-13.32 66.6-25.34-7.798 24.37-24.37 44.83-46.13 57.83 21.12-2.273 41.58-8.122 60.43-16.24-14.29 20.79-32.16 39.31-52.63 54.25z"
            fill="currentColor"
          />
        </svg>
      </div>
      <div
        class="form-starter flex flex-col justify-center px-[2rem] lg:w-[70%] lg:mx-auto "
      >
        <div
          class="icon lg:pt-[3rem] "
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-twitter text-[#1D9BF0] text-[2.5rem]"
            data-icon="twitter"
            data-prefix="fab"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M459.4 151.7c.325 4.548 .325 9.097 .325 13.65 0 138.7-105.6 298.6-298.6 298.6-59.45 0-114.7-17.22-161.1-47.11 8.447 .974 16.57 1.299 25.34 1.299 49.06 0 94.21-16.57 130.3-44.83-46.13-.975-84.79-31.19-98.11-72.77 6.498 .974 12.99 1.624 19.82 1.624 9.421 0 18.84-1.3 27.61-3.573-48.08-9.747-84.14-51.98-84.14-102.1v-1.299c13.97 7.797 30.21 12.67 47.43 13.32-28.26-18.84-46.78-51.01-46.78-87.39 0-19.49 5.197-37.36 14.29-52.95 51.65 63.67 129.3 105.3 216.4 109.8-1.624-7.797-2.599-15.92-2.599-24.04 0-57.83 46.78-104.9 104.9-104.9 30.21 0 57.5 12.67 76.67 33.14 23.72-4.548 46.46-13.32 66.6-25.34-7.798 24.37-24.37 44.83-46.13 57.83 21.12-2.273 41.58-8.122 60.43-16.24-14.29 20.79-32.16 39.31-52.63 54.25z"
              fill="currentColor"
            />
          </svg>
        </div>
        <div
          class="register-form flex flex-col "
        >
          <h2
            class="font-bold text-[4rem] py-[4rem] md:text-[2rem] sm:py-[2rem]"
          >
            Happening now
          </h2>
          <h3
            class="text-[1.7rem] font-bold"
          >
            Join Twitter today
          </h3>
          <div
            class=" cursor-pointer link my-[1rem] text-white bg-[#1D9BF0] w-[17rem] flex justify-center py-[0.5rem] rounded-[10rem] font-bold text-[0.9rem] transition-all hover:bg-[#1884cb]"
          >
            <button
              class="font-bold"
            >
              Sign up with phone or email
            </button>
          </div>
          <p
            class="text-[0.7rem] w-[17rem] leading-[0.75rem] text-[#00000096]"
          >
            By signing up, you agree to the 
            <span
              class="text-[#1D9BF0]"
            >
              Terms of Service
            </span>
             and 
            <span
              class="text-[#1D9BF0]"
            >
              Privacy Policy
            </span>
            , including 
            <span
              class="text-[#1D9BF0]"
            >
              Cookie Use.
            </span>
          </p>
        </div>
        <div
          class="login-form pt-[4rem] flex flex-col lg:pb-[2rem] sm:pt-[2rem] "
        >
          <h4
            class="font-bold text-[1.2rem]"
          >
            Already have an account?
          </h4>
          <div
            class="cursor-pointer my-[1rem] link text-[#1D9BF0] w-[17rem] flex justify-center py-[0.5rem] rounded-[10rem] font-bold text-[0.9rem] border-[1px] border-[#00000035] transition-all hover:bg-[#1883cb15]"
          >
            Sign in
          </div>
        </div>
      </div>
    </div>
    <div
      class="min-h-[10vh] text-[#7a7a7a] flex justify-center items-center px-[2rem] text-[0.9rem] w-full flex-wrap"
    >
      <p
        class="px-[0.7rem] py-[0.2rem] "
      >
        About
      </p>
      <p
        class="px-[0.7rem] py-[0.2rem] "
      >
        Help Center
      </p>
      <p
        class="px-[0.7rem] py-[0.2rem] "
      >
        Terms of Service
      </p>
      <p
        class="px-[0.7rem] py-[0.2rem] "
      >
        Privacy Policy
      </p>
      <p
        class="px-[0.7rem] py-[0.2rem] "
      >
        Cookie Policy
      </p>
      <p
        class="px-[0.7rem] py-[0.2rem] "
      >
        Accessibility
      </p>
      <p
        class="px-[0.7rem] py-[0.2rem] "
      >
        Ads info
      </p>
      <p
        class="px-[0.7rem] py-[0.2rem] "
      >
        Blog
      </p>
      <p
        class="px-[0.7rem] py-[0.2rem] "
      >
        Status
      </p>
      <p
        class="px-[0.7rem] py-[0.2rem] "
      >
        Careers
      </p>
      <p
        class="px-[0.7rem] py-[0.2rem] "
      >
        Brand Resources
      </p>
      <p
        class="px-[0.7rem] py-[0.2rem] "
      >
        Advertising
      </p>
      <p
        class="px-[0.7rem] py-[0.2rem] "
      >
        Marketing
      </p>
      <p
        class="px-[0.7rem] py-[0.2rem] "
      >
        Twitter for Business
      </p>
      <p
        class="px-[0.7rem] py-[0.2rem] "
      >
        Developers
      </p>
      <p
        class="px-[0.7rem] py-[0.2rem] "
      >
        Directory
      </p>
      <p
        class="px-[0.7rem] py-[0.2rem] "
      >
        Settings
      </p>
      <p
        class="px-[0.7rem] py-[0.2rem] "
      >
        Â© 2022 Twitter, Inc.
      </p>
    </div>
  </div>
   
</div>
`;
